### PR TITLE
SNOW-188512 added test to simulate throttling

### PR DESCRIPTION
**Pull request by sfc-gh-zli**

### CODE REVIEW

File: .github/docker/build_image.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ cd ../..
 
 # Build docker image
 docker build \
---build-arg SPARK_URL=https://apache.osuosl.org/spark/spark-3.0.0/spark-3.0.0-bin-hadoop2.7.tgz \
+--build-arg SPARK_URL=https://archive.apache.org/dist/spark/spark-3.0.0/spark-3.0.0-bin-hadoop2.7.tgz \
```
Comment (sfc-gh-zli) [inline .github/docker/build_image.sh:36]: previous link has expired

File: src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala [hunk 2]
```diff
@@ -723,6 +723,12 @@ sealed trait CloudStorage {
       "Negative test to raise error when uploading data to GCS"
     )
 
+    TestHook.raiseExceptionIfTestFlagEnabledWithMaxCount(
+      TestHookFlag.TH_UPLOAD_RAISE_EXCEPTION_WITH_COUNT,
+      "Negative test to raise error when uploading data to GCS",
+      2
+    )
```
Comment (sfc-gh-zli) [inline src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala:730]: Throw exception for the first two times, then pass.

File: src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala [hunk 3]
```diff
@@ -723,6 +723,14 @@ sealed trait CloudStorage {
       "Negative test to raise error when uploading data to GCS"
     )
 
+    // When attempt number is smaller than 2, throw exception
```
Comment (sfc-gh-mrui) [inline src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala:726]: It looks it is no necessary to keep two flags for this case. Can TH_GCS_UPLOAD_RAISE_EXCEPTION and TH_UPLOAD_RAISE_EXCEPTION_WITH_COUNT merged as one?

Comment (sfc-gh-zli) [reply src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala:716]: Merged two flags into one

**Review by sfc-gh-mrui — APPROVED** [review]
Thanks for the good work!

**Pull request merged**